react/src/components/Widget/Widget.tsx

```diff
@@ -263,46 +262,50 @@ export const Widget: React.FC<WidgetProps> = props => {
     }
     url = prefixedAddress + (query.length ? `?${query.join('&')}` : '');
 
+    let addressType: currency = 'AUD';
+
+    if (isValidCashAddress(address)) {
+      addressType = 'BCH';
+      setWidgetButtonText(`Send with BCH wallet`);
+    } else if (isValidXecAddress(address)) {
+      addressType = 'XEC';
+      setWidgetButtonText(`Send with XEC wallet`);
+    }
```

Comment:
I think you can leverage the `validCurrencyType` function to avoid if statements here.

```js
const currency = validCurrencyType(address);
setWidgetButtonText(`Send with ${currency} wallet`);
```

Reply:
done